scripts/hooks/pre-commit-migration-collision.sh

```
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Migration Timestamp Collision Check
+# Prevents committing migration files with duplicate timestamps.
+# Add to pre-commit hook chain or run standalone.
+
+MIGRATION_DIR="supabase/migrations"
+
+if git diff --cached --name-only | grep -q "^${MIGRATION_DIR}/"; then
+    STAGED_MIGRATIONS=$(git diff --cached --name-only | grep "^${MIGRATION_DIR}/" | sed "s|${MIGRATION_DIR}/||" | sed 's/_.*//' | sort)
+    EXISTING_MIGRATIONS=$(ls "$MIGRATION_DIR" 2>/dev/null | sed 's/_.*//' | sort)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)